src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -332,20 +337,22 @@ internal Location CreateLocation(LocationVersionOne v1Location)
                 {
                     FullyQualifiedLogicalName = v1Location.LogicalLocationKey ?? v1Location.FullyQualifiedLogicalName,
                     PhysicalLocation = CreatePhysicalLocation(v1Location.ResultFile),
-                    Properties = v1Location.Properties
+                    Properties = v1Location.Properties,
                 };
 
                 if (!string.IsNullOrWhiteSpace(location.FullyQualifiedLogicalName))
                 {
-                    if (_currentRun.LogicalLocations?.ContainsKey(location.FullyQualifiedLogicalName) == true)
+                    if (_v1KeyToV2LogicalLocationMap.TryGetValue(location.FullyQualifiedLogicalName, out LogicalLocation logicalLocation))
                     {
-                        _currentRun.LogicalLocations[location.FullyQualifiedLogicalName].DecoratedName = v1Location.DecoratedName;
-                    }
-                    else
-                    {
-                        LogicalLocation logicalLocation = CreateLogicalLocation(location.FullyQualifiedLogicalName,
-                                                                                decoratedName: v1Location.DecoratedName);
-                        location.FullyQualifiedLogicalName = AddLogicalLocation(logicalLocation);
+                        _logicalLocationToIndexMap.TryGetValue(logicalLocation, out int index);
+
+                        if (!string.IsNullOrEmpty(logicalLocation.DecoratedName))
+                        {
+                            logicalLocation.DecoratedName = v1Location.DecoratedName;
+                            _logicalLocationToIndexMap[logicalLocation] = index;
```

Comment:
>index [](start = 74, length = 5)

I don't get why the population of `_logicalLocationToIndexMap` depends on whether there's a decorated name. If this code is correct, it needs a comment.

Reply:
something looks odd here. also note that when we can't retrieve the logical location, we're setting the index to 0 which is wrong. 

[https://github.com/Microsoft/sarif-sdk/issues/1183](https://github.com/Microsoft/sarif-sdk/issues/1183)

---
In reply to: [243868420](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243868420) [](ancestors = 243868420)